ValidationLibrary.AzureFunctions/StatusEndpoint.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace ValidationLibrary.AzureFunctions
+{
+    public class StatusEndpoint
+    {
+        private readonly ILogger<StatusEndpoint> _logger;
+        private ValidationLibrary.RepositoryValidator _validator;
+        public StatusEndpoint(ILogger<StatusEndpoint> logger, ValidationLibrary.RepositoryValidator validator)
+        {
+            _logger = logger;
+            _validator = validator;
+        }
+
+        [FunctionName("StatusCheck")]
+        public IActionResult Run([HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequestMessage req)
+        {
+            try
+            {
+                _logger.LogDebug("Repository validator status check hook launched.");
+
+                return new JsonResult(new {
+                    Rules = _validator.GetRules().Select(r => r.GetConfiguration())
+                });
+            }
+            catch (Exception exception)
+            {
+                if (exception is ArgumentException || exception is JsonException)
+                {
+                    _logger.LogError(exception, "Invalid request received");
+
+                    return new BadRequestResult();
```

Comment:
OK, I think you mean like this.

Reply:
> 
> 
> I don't this should ever throw `BadRequestResult` because there aren't any parameters. All errors in this should just throw 500 or related, so no need for error handling.

Actually, after looking at how simple it is, I have removed the try/catch, maybe that's what you wanted anyway. The Run method is so simple and it doesn't pass execution elsewhere/have other dependencies which is normally the situation when try/catch is used. I can put it back though if you want.

Reply:
Yeah, that's what I meant. I think those exceptions are automatically logged.